build-tools/packages/build-cli/src/commands/generate/assertTags.ts

```diff
@@ -97,71 +159,120 @@ export class TagAssertsCommand extends PackageCommand<typeof TagAssertsCommand>
 		}
 	}
 
-	protected async processPackage<TPkg extends Package>(
+	// This should not be called due to processPackages being overridden instead.
+	protected override async processPackage<TPkg extends Package>(
 		pkg: TPkg,
 		kind: PackageKind,
 	): Promise<void> {
-		const tsconfigPath = await this.getTsConfigPath(pkg);
-		this.collectAssertData(tsconfigPath);
+		throw new Error("Method not implemented.");
 	}
 
-	public async run(): Promise<void> {
-		// Calls processPackage on all packages to collect assert data.
-		await super.run();
+	protected override async processPackages(packages: PackageWithKind[]): Promise<string[]> {
+		const errors: string[] = [];
 
-		// Tag asserts based on earlier collected data.
-		this.tagAsserts(true);
-	}
+		const collected: CollectedData = {
+			shortCodes: new Map<number, Node>(),
+			codeToMsgMap: new Map<string, string>(),
+			maxShortCode: -1,
+		};
+
+		const dataMap = new Map<PackageWithKind, PackageData>();
+		const config = cosmiconfig("assertTagging");
+
+		for (const pkg of packages) {
+			// Package configuration:
+			// eslint-disable-next-line no-await-in-loop
+			const tsconfigPath = await this.getTsConfigPath(pkg);
+			// eslint-disable-next-line no-await-in-loop
+			const packageConfig = await config.search(pkg.directory);
```

Comment:
If you only support one format, you can use .load instead of .search, which is a little faster. But if you want/need to support both CJS and ESM configs, this is fine.

Reply:
Not quite. We leverage the parent directory search functionality (ex: I didn't configure ever package in packages/test separately) and that requires using "search".

If we just wanted a single format from a single location, I'd just use `readJson` and avoid pulling in consmic config and all its complexity. However since we already depend on cosmic config, might as well benefit from it.